### PR TITLE
A lot of code improvements

### PR DESCRIPTION
**Pull request by darxriggs**
The code has been improved in different aspects.

Each commit contains a different aspect. Therefore I suggest that this pull request is reviewed commit by commit.

The user-facing functionality has also slightly been improved:
* trim inputs
* consistently handle settings files (.xml) with UTF-8 encoding
* consistently handle properties files (.properties) with ISO-8859-1 encoding

### CODE REVIEW

**Comment by darxriggs** [comment]
@olivergondza / @rjclark / @gboissinot is there anything else I can do to get this merged?

**Comment by olivergondza** [comment]
Sorry, I am not the one to review this.

**Comment by darxriggs** [comment]
@rjclark / @gboissinot can you have a look at this?

I want to fix [JENKINS-53570](https://issues.jenkins-ci.org/browse/JENKINS-53570) after this has been merged.

**Comment by darxriggs** [comment]
@rjclark / @gboissinot ping

**Comment by darxriggs** [comment]
@rjclark / @gboissinot let me try again.

**Comment by TonyNoble** [comment]
Hi @darxriggs 

I've recently taken on maintenance of this plugin, so I'll take a look as soon as possible and get the PR merged.  Please bear with me, the XTrigger-related plugins are all a little dusty around the edges :)

**Comment by darxriggs** [comment]
@TonyNoble thanks for stepping in
